core/src/main/scala/com/cloudera/spark/CGroupsMemoryHandle.scala

```diff
@@ -0,0 +1,56 @@
+package com.cloudera.spark
+
+import com.cloudera.spark.CGroupsMemoryHandle.{cGroupMemoryStatPath, getCGroupMemStats}
+
+import scala.io.Source
+
+class CGroupsMemoryHandle(cgroupMemStats: Seq[(String, Long)]) extends MemoryGetter {
+
+  override val namesAndReporting: Seq[(String, PeakReporting)] = cgroupMemStats.map(x =>
+    if (Set("pgpgin", "pgpgout", "total_pgpgin", "total_pgpgout")(x._1)) {
+      (x._1, IncrementCounts)
+    } else {
+      (x._1, IncrementBytes)
+    }
+  )
+
+  override def values(dest: Array[Long], offset: Int): Unit = {
+    if (new java.io.File(cGroupMemoryStatPath).exists) {
+      var counter = 0
+      getCGroupMemStats.foreach{ case (_, metricValue) =>
```

Comment:
style nit: space after `foreach`

this is still assuming the metrics come in a consistent order in that file, isn't it?  If the order happens to change, you'll get garbage updates?

Reply:
done for the style nit.
Removing the assumption about the metric ordering in the latest commit. Latest commit strictly enforces a fixed ordering.